### PR TITLE
Add remaining k8s endpoints for Folders

### PR DESCRIPTION
**Pull request by filewalkwithme**
**What is this feature?**

Add 3 more k8s endpoints for Folders:
- `GET /` : List Folders
- `PUT /:uid/`: Update Folders
- `DELETE /:uid/`: Delete Folders

**Why do we need this feature?**

This is a requirement for onboarding Folders on UniStore

**Who is this feature for?**

N/A

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/search-and-storage-team/issues/84

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/api/folder.go [hunk 1]
```diff
@@ -648,17 +649,43 @@ func newFolderK8sHandler(hs *HTTPServer) *folderK8sHandler {
...
+	c.JSON(http.StatusOK, folders)
+}
+
 func (fk8s *folderK8sHandler) createFolder(c *contextmodel.ReqContext) {
 	client, ok := fk8s.getClient(c)
 	if !ok {
 		return // error is already sent
 	}
-	cmd := folder.CreateFolderCommand{}
+	cmd := folder.UpdateFolderCommand{}
```
Comment (suntala) [inline pkg/api/folder.go:683]: Do we need to replace this? I saw that Playlists use `UpdatePlaylistCommand` but I am not sure that we do.

Comment (filewalkwithme) [reply pkg/api/folder.go:683]: Fixed

**Comment by filewalkwithme** [comment]
@suntala I have addressed the comments. Could you take another look when you have some time?

**Review by suntala — APPROVED** [review]
LGTM

**Pull request merged**